src/main/java/me/jellysquid/mods/sodium/client/world/WorldSlice.java

```diff
@@ -306,10 +313,18 @@ private void unpackBlockDataZ(Block[] blocks, int[] metas, ClonedChunkSection se
     }
 
     public Block getBlock(int x, int y, int z) {
+        if (!blockBoxContains(this.volume, x, y, z)) {
+            return Blocks.air;
+        }
+
         final int relX = x - this.baseX;
         final int relY = y - this.baseY;
         final int relZ = z - this.baseZ;
-        return this.blockArrays[getLocalSectionIndex(relX >> 4, relY >> 4, relZ >> 4)][getLocalBlockIndex(relX & 15, relY & 15, relZ & 15)];
+        try {
+            return this.blockArrays[getLocalSectionIndex(relX >> 4, relY >> 4, relZ >> 4)][getLocalBlockIndex(relX & 15, relY & 15, relZ & 15)];
+        } catch(IndexOutOfBoundsException ignored) {
```

Comment:
Do not catch IOOB here, it should never be thrown if `blockBoxContains` is implemented correctly.

Reply:
oh yah, this was put in before I added the `blockBoxContains` check as I was debugging things